### PR TITLE
chore(clangd): add auto-path completion for query-driver

### PR DESCRIPTION
**Pull request by aarnphm**
I added this small change for clangd to auto-complete the tools that is used for --query-driver

Maybe after custom config for users mature a bit more, we can allow them to configure such features as well, but for now if users need to add any new driver they can just add it in the table instead of modifying the specific path.

Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

**Comment by ayamir** [comment]
So the `clang++` which is not in the `PATH` can't be found in this way. Users should add where the `clang++` located to `PATH`.

**Comment by aarnphm** [comment]
Hmm. Maybe we can also inform users to add such driver into PATH in the Wiki? What do you think?

File: lua/modules/configs/completion/servers/clangd.lua [hunk 1]
```diff
@@ -22,6 +23,30 @@ local function switch_source_header_splitcmd(bufnr, splitcmd)
 	end
 end
 
+local function get_binary_path(binary)
+	local path = nil
+	if global.is_mac or global.is_linux then
+		path = vim.fn.trim(vim.fn.system("which " .. binary))
+	elseif global.is_windows then
+		path = vim.fn.trim(vim.fn.system("where " .. binary))
```
Comment (aarnphm) [inline lua/modules/configs/completion/servers/clangd.lua:31]: Also from the Windows folks my question is whether this is the correct way to find binary?
I don't have a Windows machine so I'm not entirely sure

Comment (ayamir) [reply lua/modules/configs/completion/servers/clangd.lua:31]: Test passwd for window.

**Comment by ayamir** [comment]
> Hmm. Maybe we can also inform users to add such driver into PATH in the Wiki? What do you think?

OK. This commit also consider the problem caused by different OS. It's also intuitive that `clang++` should in the `PATH` for most users. but we also need consider users who just use multiple versions of `clang++` and note for them.

**Comment by aarnphm** [comment]
I can update the Wiki if needed.

**Review by ayamir — APPROVED** [review]

**Comment by CharlesChiuGit** [comment]
> I can update the Wiki if needed.

Tks! U can type what needs to added, and i'll add it later.

**Comment by aarnphm** [comment]
ok. So I think we just need to inform users who use clangd to add any of the required `query-driver` to PATH on all platform.

**Comment by CharlesChiuGit** [comment]
Do u have a draft that I can work on?

**Comment by aarnphm** [comment]
I don't have one right now. I'm wondering if the repo should include a FAQ section 🤔

**Comment by CharlesChiuGit** [comment]
https://github.com/ayamir/nvimdots/wiki/Issues

**Comment by aarnphm** [comment]
cc @CharlesChiuGit https://github.com/ayamir/nvimdots/wiki/Issues#issues-while-using-clangd-cc-lsp

**Comment by CharlesChiuGit** [comment]
Nice! So u have permissions to edit wikl?

**Comment by ayamir** [comment]
> Nice! So u have permissions to edit wikl?

Everyone has the edit permission for convenience.

**Comment by aarnphm** [comment]
Oh, is that for contributor only? or anyone from the internet can edit?

if it is the latter then ... 😄

**Pull request merged**